.kokoro/build-with-run.sh

```diff
@@ -32,6 +32,44 @@ popd
 export GOOGLE_CLOUD_PROJECT=nodejs-docs-samples-tests
 pushd github/nodejs-docs-samples/${PROJECT}
 
+# Verify changes worth testing.
+SIGNIFICANT_CHANGES="$(git --no-pager diff --name-only main..HEAD | grep -Ev '(\.md$|^\.github)' || true)"
```

Comment:
**Thought (feel free to ignore):**
I wonder if this `.kokoro` script (`build-with-run.sh`) has access to a variable that contains the name of the target branch (so we're not stuck to comparing to `main`).